File: ScaleformUI_Csharp/Windows/UIMenuDetailsWindow.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScaleformUI
{
    public class UIMenuDetailsWindow : UIMenuWindow
    {
        public string DetailTop;
        public string DetailMid;
        public string DetailBottom;
        public UIDetailImage DetailLeft;
        public bool StatWheelEnabled;
        public List<UIDetailStat> DetailStats;
        public UIMenuDetailsWindow(string top, string mid, string bot, UIDetailImage leftDetail = null)
        {
            id = 1;
            DetailTop = top;
            DetailMid = mid;
            DetailBottom = bot;
            StatWheelEnabled = false;
            DetailLeft = leftDetail ?? new();
        }

        public UIMenuDetailsWindow(string top, string mid, string bot, bool statWheelEnabled, List<UIDetailStat> details)
        {
            id = 1;
            DetailTop = top;
            DetailMid = mid;
            DetailBottom = bot;
            StatWheelEnabled = statWheelEnabled;
            DetailStats = details;
            DetailLeft = new()
            {
                Txd = "statWheel"
            };
        }

        public void UpdateLabels(string top, string mid, string bot, UIDetailImage leftDetail = null)
        {
            DetailTop = top;
            DetailMid = mid;
            DetailBottom = bot;
            DetailLeft = leftDetail ?? new();
            if (ParentMenu is not null)
            {
                var wid = ParentMenu.Windows.IndexOf(this);
                if (!StatWheelEnabled)
                    ScaleformUI._ui.CallFunction("UPDATE_DETAILS_WINDOW_VALUES", wid, DetailBottom, DetailMid, DetailTop, DetailLeft.Txd, DetailLeft.Txn, DetailLeft.Pos.X, DetailLeft.Pos.Y, DetailLeft.Size.Width, DetailLeft.Size.Height);
                else
                    ScaleformUI._ui.CallFunction("UPDATE_DETAILS_WINDOW_VALUES", wid, DetailBottom, DetailMid, DetailTop, "statWheel");
            }
        }

        public void AddStatsListToWheel(List<UIDetailStat> stats)
        {
            if (StatWheelEnabled)
            {
                DetailStats = stats;
                if (ParentMenu is not null)
                {
                    var wid = ParentMenu.Windows.IndexOf(this);
                    foreach (var value in stats)
                    {
                        ScaleformUI._ui.CallFunction("ADD_STATS_DETAILS_WINDOW_STATWHEEL", wid, value.Percentage, (int)value.HudColor);
                    }
                }
            }
        }

        public void AddStatSingleToWheel(UIDetailStat stat)
        {
            if (StatWheelEnabled)
            {
                DetailStats.Add(stat);
                if (ParentMenu is not null)
                {
                    var wid = ParentMenu.Windows.IndexOf(this);
                    ScaleformUI._ui.CallFunction("ADD_STATS_DETAILS_WINDOW_STATWHEEL", wid, stat.Percentage, (int)stat.HudColor);
                }
            }
        }

        public void UpdateStatsToWheel()
        {
            UpdateStatsToWheel(DetailStats);
        }
        public void UpdateStatsToWheel(List<UIDetailStat> stats)
        {
            if (StatWheelEnabled)
            {
                if(DetailStats.Count != stats.Count)
                {
                    throw new Exception("You cannot add items using this function");
                }
                DetailStats = stats;
                if (ParentMenu is not null)
                {
                    var wid = ParentMenu.Windows.IndexOf(this);
                    foreach (var value in DetailStats)
                    {
                        ScaleformUI._ui.CallFunction("UPDATE_STATS_DETAILS_WINDOW_STATWHEEL", wid, DetailStats.IndexOf(value), value.Percentage, (int)value.HudColor);
                    }
                }
            }
        }
        public void RemoveStatToWheel(UIDetailStat stat)
        {
            var s = DetailStats.FirstOrDefault(x => x.Percentage == stat.Percentage && x.HudColor == stat.HudColor);
            if (s is not null)
            {
                RemoveStatToWheel(DetailStats.IndexOf(s));
            }
        }
        public void RemoveStatToWheel(int id)
        {
            if (id < 0 || id >= DetailStats.Count) return;
            DetailStats.RemoveAt(id);
            if(ParentMenu is not null)
            {
                var wid = ParentMenu.Windows.IndexOf(this);
                ScaleformUI._ui.CallFunction("REMOVE_STATS_DETAILS_WINDOW_STATWHEEL", wid, id);
            }
        }
    }

    public class UIDetailImage
    {
        public string Txd { get; set; }
        public string Txn { get; set; }
        public PointF Pos { get; set; }
        public SizeF Size { get; set; }

        public UIDetailImage()
        {
            Txd = "";
            Txn = "";
            Pos = new(0, 0);
            Size = new(0, 0);
        }
        public UIDetailImage(string txd, string txn, PointF pos, SizeF size)
        {
            Txd = txd;
            Txn = txn;
            Pos = pos;
            Size = size;
        }
    }

    public class UIDetailStat
    {
        public int Percentage;
        public HudColor HudColor;
        public UIDetailStat(int percentage, HudColor color)
        {
            Percentage = percentage;
            HudColor = color;
        }
    }
}

File: ScaleformUI_Lua/src/windows/UIMenuDetailsWindow.lua
UIMenuDetailsWindow = setmetatable({}, UIMenuDetailsWindow)
UIMenuDetailsWindow.__index = UIMenuDetailsWindow
UIMenuDetailsWindow.__call = function() return "UIMenuWindow", "UIMenuDetailsWindow" end

---New
---@param Mom number
---@param Dad number
function UIMenuDetailsWindow.New(...)
    local args = {...}

    if #args == 3 or #args == 4 then
        _UIMenuDetailsWindow = {
            id = 1,
            DetailTop = args[1],
            DetailMid = args[2],
            DetailBottom = args[3],
            StatWheelEnabled = false,
            DetailLeft = args[4] or {
                Txd = "",
                Txn = "",
                Pos = vector2(0, 0),
                Size = vector2(0, 0),
            },
            ParentMenu = nil, -- required
        }
    elseif #args == 5 then
        _UIMenuDetailsWindow = {
            id = 1,
            DetailTop = args[1],
            DetailMid = args[2],
            DetailBottom = args[3],
            StatWheelEnabled = args[4],
            DetailStats = args[5],
            DetailLeft = {
                Txd = "statWheel",
                Txn = "",
                Pos = vector2(0, 0),
                Size = vector2(0, 0),
            },
            ParentMenu = nil, -- required
        }
    end
	return setmetatable(_UIMenuDetailsWindow, UIMenuDetailsWindow)
end

---SetParentMenu
---@param Menu table
function UIMenuDetailsWindow:SetParentMenu(Menu) -- required
	if Menu() == "UIMenu" then
		self.ParentMenu = Menu
	else
		return self.ParentMenu
	end
end

function UIMenuDetailsWindow:UpdateLabels(top, mid, bot, leftDetail)
    self.DetailTop = top
    self.DetailMid = mid
    self.DetailBottom = bot
    self.DetailLeft = leftDetail or {
        Txd = "",
        Txn = "",
        Pos = vector2(0, 0),
        Size = vector2(0, 0),
    }

    if self.ParentMenu ~= nil then
        local wid = IndexOf(self.ParentMenu.Windows, self) - 1
        if self.StatWheelEnabled then
            ScaleformUI.Scaleforms._ui:CallFunction("UPDATE_DETAILS_WINDOW_VALUES", false, wid, self.DetailBottom, self.DetailMid, self.DetailTop, "statWheel")
        else
            ScaleformUI.Scaleforms._ui:CallFunction("UPDATE_DETAILS_WINDOW_VALUES", false, wid, self.DetailBottom, self.DetailMid, self.DetailTop, self.DetailLeft.Txd, self.DetailLeft.Txn, self.DetailLeft.Pos.x, self.DetailLeft.Pos.y, self.DetailLeft.Size.x, self.DetailLeft.Size.y)
        end
    end
end

function UIMenuDetailsWindow:AddStatsListToWheel(stats)
    if self.StatWheelEnabled then
        self.DetailStats = stats
        if self.ParentMenu ~= nil then
            local wid = IndexOf(self.ParentMenu.Windows, self) - 1
            for key, value in pairs(self.DetailStats) do
                ScaleformUI.Scaleforms._ui:CallFunction("ADD_STATS_DETAILS_WINDOW_STATWHEEL", false, wid, value.Percentage, value.HudColor)
            end
        end
    end
end

function UIMenuDetailsWindow:AddStatSingleToWheel(stat)
    if self.StatWheelEnabled then
        self.DetailStats[#self.DetailStats + 1] = stat
        if self.ParentMenu ~= nil then
            local wid = IndexOf(self.ParentMenu.Windows, self) - 1
            ScaleformUI.Scaleforms._ui:CallFunction("ADD_STATS_DETAILS_WINDOW_STATWHEEL", false, wid, stat.Percentage, stat.HudColor)
        end
    end
end

function UIMenuDetailsWindow:UpdateStatsToWheel()
    if self.StatWheelEnabled then
        if self.ParentMenu ~= nil then
            local wid = IndexOf(self.ParentMenu.Windows, self) - 1
            for key, value in pairs(self.DetailStats) do
                ScaleformUI.Scaleforms._ui:CallFunction("UPDATE_STATS_DETAILS_WINDOW_STATWHEEL", false, wid, key - 1, value.Percentage, value.HudColor)
            end
        end
    end
end